### PR TITLE
chore: Include env variable copying into yarn setup command

### PR DESCRIPTION
**Pull request by smilingkylan**
## **Description**

This task is an update to the on-boarding process, helping streamline the creation of environment variables.

## **Related issues**

Works on: https://github.com/MetaMask/mobile-planning/issues/1465

## **Manual testing steps**

1. Delete (or change the file name) of `.ios.env`, `.android.env`, and `.js.env` files. Then run `yarn setup` and see if new files are created with default values.

## **Screenshots/Recordings**

https://github.com/MetaMask/metamask-mobile/assets/6249205/f169d46f-9057-441a-ba3a-60fcb45528cb

### **Before**

Before you had to manually use a `cp ...` command, now it is included in `yarn setup` **if the files do not already exist**

### **After**

Now the `yarn setup` command generates the `.env` files. See earlier screen recording.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've clearly explained what problem this PR is solving and how it is solved.
- [X] I've linked related issues
- [X] I've included manual testing steps
- [X] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by gauthierpetetin — DISMISSED** [review]

File: ios/Podfile.lock [hunk 1]
```diff
@@ -901,7 +901,7 @@ EXTERNAL SOURCES:
 
 SPEC CHECKSUMS:
   Base64: cecfb41a004124895a7bcee567a89bae5a89d49b
-  boost: 57d2868c099736d80fcd648bf211b4431e51a558
+  boost: 7dcd2de282d72e344012f7d6564d024930a6a440
```
Comment (tommasini) [inline ios/Podfile.lock:904]: I don't have this change on my side, but I saw it on others machines, do we know why this happen?

**Review by jpcloureiro — DISMISSED** [review]
lgtm

Great addition Kylan 👍

File: scripts/postinstall.sh [hunk 2]
```diff
@@ -22,3 +22,17 @@ echo "" > ios/release.xcconfig
 echo "6. Init git submodules"
 echo "This may take a while..."
 git submodule update --init
+
+echo "7. Copy env variables..."
```
Comment (NicolasMassart) [inline scripts/postinstall.sh:26]: Maybe add a message to remind to customise the values in these newly created env files?

Comment (jpcloureiro) [reply scripts/postinstall.sh:26]: I would prefer to have that kind of information in a different place, such as a wiki or FAQ for maintainers.
Usually, no one bothers to read `yarn setup` logs unless it fails.

File: ios/Podfile.lock (hunk 1 above)
Comment (jpcloureiro) [reply ios/Podfile.lock:904]: Yes we can discard this change.
These checksums can change due to different factors.

Here's a [discussion on react-native](https://github.com/facebook/react-native/issues/31121) repo about it

**Review by jpcloureiro — APPROVED** [review]

**Pull request merged**